packages/idom/src/lib/idom.js

```diff
@@ -46,7 +46,13 @@ function handleChildren(children, parentNode) {
         return false;
     }
 
-    text(children);
+    if (parentNode.__dnaTextContent !== children) {
+        text(children);
+        parentNode.__dnaTextContent = children;
+    } else {
+        skip();
+    }
```

Comment:
I would add this check in order to improve reliability

```suggestion
    if (parentNode.__dnaTextContent !== children || parentNode.children.length || parentNode.childNodes.length > 1) {
        text(children);
        parentNode.__dnaTextContent = children;
    } else {
        skip();
    }
```